includes/class-gp-translation-helpers.php

```diff
@@ -411,24 +411,31 @@ function( $reason ) use ( $all_reject_reasons ) {
 	 * @since 0.0.2
 	 *
 	 *  @param string $reject_comment Feedback entered by reviewer.
-	 *  @param int    $post_id ID of the post where the comment will be added.
-	 *  @param array  $reject_reason Reason(s) for rejection.
+	 *  @param int    $post_id        ID of the post where the comment will be added.
+	 *  @param array  $reject_reason  Reason(s) for rejection.
 	 *  @param string $translation_id ID of the rejected translation.
-	 *  @param string $locale_slug Locale of the rejected translation.
+	 *  @param string $locale_slug    Locale of the rejected translation.
+	 *  @param array  $server         The $_SERVER array
 	 *
 	 * @return false|int
 	 */
-	private function insert_reject_comment( $reject_comment, $post_id, $reject_reason, $translation_id, $locale_slug ) {
+	private function insert_reject_comment( $reject_comment, $post_id, $reject_reason, $translation_id, $locale_slug, $server ) {
+		$user = wp_get_current_user();
 		return wp_insert_comment(
 			array(
-				'comment_content' => $reject_comment,
-				'comment_post_ID' => $post_id,
-				'comment_meta'    => array(
+				'comment_post_ID'      => $post_id,
+				'comment_author'       => $user->display_name,
+				'comment_author_email' => $user->user_email,
+				'comment_author_url'   => $user->user_url,
+				'comment_author_IP'    => sanitize_text_field( $server['REMOTE_ADDR'] ),
```

Comment:
Would this not be the same if we had used `$_SERVER['REMOTE_ADDR']`? Not sure we need to pass `$server` to this function.

Reply:
In this situation, we are strongly coupled between the function and the $_SERVER array, but I like to pass the data to the functions, using [dependency injection](https://en.wikipedia.org/wiki/Dependency_injection). In fact, it will be a good practice to do the same with the `$user` used in the function.